### PR TITLE
[test-framework] Add health check to local envoy agent at startup

### PR DESCRIPTION
**Pull request by nmittler**
Also fixing a bug in the envoy yaml template that was preventing the
configuration of multiple ports to a single instance.  Expanded the
test to cover this.

### CODE REVIEW

File: pkg/test/proxy/envoy/envoy.go [hunk 1]
```diff
@@ -104,10 +142,32 @@ func (e *Envoy) Start() error {
...
-	if err := e.cmd.Start(); err != nil {
-		e.returnBaseID()
+	if err = e.cmd.Start(); err != nil {
 		return err
 	}
+
+	endTime := time.Now().Add(startTimeout)
+	for {
+		// Sleep a short before attempting to connect to Envoy.
+		time.Sleep(1 * time.Second)
```
Comment (ozevren) [inline pkg/test/proxy/envoy/envoy.go]: Why do we need this sleep? To let Envoy state to quiesce? 1 second granularity seems to high (I've been experimenting with sleep/retry loops when doing the Docker based model, and I've found that 1 second delay to be a significant source of startup delay. 

Instead, can we do a "reasonable" initial delay, coupled with a short retry delay? How quick does the GetServerInfo return if there is an error?

**Comment by nmittler** [comment]
@ozevren PTAL

**Comment by nmittler** [comment]
@ozevren also note that in my last commit, I moved the proxy code to the "local" package ... I think that makes more sense.

**Pull request merged**